python/jiminy_py/src/jiminy_py/log.py

```diff
@@ -217,9 +231,67 @@ def build_robot_from_log(
 
     return robot
 
+def build_robots_from_log(
+        log_data: Dict[str, Any],
+        mesh_path_dir: Optional[str] = None,
+        mesh_package_dirs: Sequence[str] = (),
+        ) -> Sequence[jiminy.Robot]:
+    """Build robots from log. For mulit-robot simulations, build all the robots
+    in the log of the simulation.
+
+    .. note::
+        model options and `robot.pinocchio_model` will be the same as during
+        the simulation until the next call to `reset` method unless the options
+        of the robots that has been restored are overwritten manually.
+
+    .. note::
+        It returns a sequence of valid and fully initialized robots, that can
+        be used to perform new simulation if added to a Jiminy Engine, but the
+        original controller is lost.
+
+    .. warning::
+        It does ot require the original URDF files to exist, but the original
+        mesh paths (if any) must be valid since they are not bundle in the log
+        archive for now.
+
+    :param log_data: Logged data (constants and variables) as a dictionary.
+    :param mesh_path_dir: Overwrite the common root of all absolute mesh paths.
+                          It which may be necessary to read log generated on a
+                          different environment.
+    :param mesh_package_dirs: Prepend custom mesh package search path
+                              directories to the ones provided by log file. It
+                              may be necessary to specify it to read log
+                              generated on a different environment.
+
+    :returns: Sequence of reconstructed robots.
+    """
+    # Extract log constants
+    log_constants = log_data["constants"]
+
+    robots_names = []
+    
+    for key in log_constants.keys():
+        if key == "HighLevelController.has_freeflyer":
+            robots_names.append("")
+        else:
+            m = re.findall(r"HighLevelController.(\w+).has_freeflyer", key)
+            if len(m) == 1:
+                robots_names.append(m[0])
+    
+    robots = []
+
+    for robot_name in robots_names:
+        robot = build_robot_from_log(log_data, mesh_path_dir,
+                                     mesh_package_dirs, robot_name=robot_name)
+
+        robots.append(robot)
+
+    return robots
+
 
 def extract_trajectory_from_log(log_data: Dict[str, Any],
-                                robot: Optional[jiminy.Model] = None
+                                robot: Optional[jiminy.Model] = None,
+                                *, robot_name: str = ""
```

Comment:
No need for `robot_name` if the robot has been specified, but you need it if it was not...
So I would suggest to change the signature for this: `robot_name: Optional[str] = None`, and use `robot.name` by default if specified, or `""` if not and `robot_name` is not specified either, and finally `robot_name` if specified by not robot. If both are specified, just raise an exception.